enos/modules/set_undo_logs_env_var/main.tf

```diff
@@ -0,0 +1,16 @@
+terraform {
+  required_providers {
+    enos = {
+      source = "app.terraform.io/hashicorp-qti/enos"
+    }
+  }
+}
+
+variable "undo_logs" {
+  type        = string
+  description = "The variable takes either 0 or 1 to indicate disabling or enabling undo logs"
+}
+
+resource "enos_local_exec" "set_undo_logs_env_var" {
+  inline = ["export VAULT_REPLICATION_USE_UNDO_LOGS=${var.undo_logs}"]
```

Comment:
Does this need to be set before the autopilot upgrade step ?  The environment variable won't stay between steps as each makes its own instance connection, so you'll need to set the variable in the module/step where it is used